qsimcirq/qsim_circuit.py

```diff
@@ -60,6 +60,10 @@ def translate_cirq_to_qsim(
     circuit_data = []
     ordered_qubits = cirq.ops.QubitOrder.as_qubit_order(qubit_order).order_for(
         self.all_qubits())
+
+    # qsim numbers qubits in reverse order from cirq
+    ordered_qubits = list(reversed(ordered_qubits))
```

Comment:
With this line commented out, the full-state test still passes. The culprit appears to be these lines:
https://github.com/quantumlib/qsim/blob/88f187234645c92a8a9bc77ad6f425b24644dcb0/qsimcirq/qsim_simulator.py#L151-L153

Note that `ordered_qubits` is (incorrectly) defined using the pre-translation version of the circuit.